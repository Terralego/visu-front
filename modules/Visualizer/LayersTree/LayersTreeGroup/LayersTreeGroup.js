import React from 'react';
import { H5, Button, Collapse } from '@blueprintjs/core';

import LayersTreeItem from '../LayersTreeItem';

export class LayersTreeGroup extends React.Component {
  // Get open from layer if not set default value
  // eslint-disable-next-line
  state = this.props.layer.initialState || {
    open: true,
  };

  handleClick = () => {
    const { open } = this.state;
    this.setState({ open: !open });
  };

  render () {
    const { open } = this.state;
    const {
      title,
      layer: { layers },
      isHidden,
    } = this.props;

    const { handleClick } = this;
    return (
      isHidden ? null : (
        <div className="layerstree-group">
          <div className="layerstree-group__label">
            <Button
              className="layerstree-group__label-button"
              onClick={handleClick}
              icon={open ? 'chevron-down' : 'chevron-right'}
              minimal
            />
            <H5>{title}</H5>
          </div>
          <Collapse
            isOpen={open}
          >
            {layers.map(layer => (
              <LayersTreeItem
                key={layer.label}
                layer={layer}
              />
            ))}
          </Collapse>
        </div>
      )
    );
  }
}

export default LayersTreeGroup;
