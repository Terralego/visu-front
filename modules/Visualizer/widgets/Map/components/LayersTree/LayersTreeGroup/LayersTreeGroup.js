import React from 'react';
import { H4 } from '@blueprintjs/core';

import LayersTreeItem from '../LayersTreeItem';

export const LayersTreeGroup = ({ title, layers }) => (
  <div className="layers-tree-group">
    <H4>{title}</H4>
    {layers.map(layer => (
      <LayersTreeItem
        key={layer.label}
        layer={layer}
      />
    ))}
  </div>
);

export default LayersTreeGroup;
