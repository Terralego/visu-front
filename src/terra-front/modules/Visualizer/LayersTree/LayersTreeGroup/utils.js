export const isGroupHidden = (layers, getLayerState) => layers.reduce((prev, layer) =>
  (prev && (
    layer.layers
      ? (getLayerState({ layer }).hidden ||
        isGroupHidden(layer.layers, getLayerState))
      : getLayerState({ layer }).hidden
  )),
true);

export default { isGroupHidden };
