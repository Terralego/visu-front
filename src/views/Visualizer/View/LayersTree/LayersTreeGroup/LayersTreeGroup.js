import React from 'react';
import { H5 } from '@blueprintjs/core';

import LayersTreeItem from '../LayersTreeItem';

export const LayersTreeGroup = ({
  title,
  layers,
  private: isPrivate,
  authenticated,
}) => ((!isPrivate || authenticated)
  ? (
    <div className="layers-tree-group">
      <H5>{title}</H5>
      {layers.map(layer => (
        <LayersTreeItem
          key={layer.label}
          layer={layer}
        />
      ))}
    </div>
  )
  : null);

export default LayersTreeGroup;
