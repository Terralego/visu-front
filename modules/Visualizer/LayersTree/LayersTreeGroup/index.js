import LayersTreeGroup from './LayersTreeGroup';

export default LayersTreeGroup;
