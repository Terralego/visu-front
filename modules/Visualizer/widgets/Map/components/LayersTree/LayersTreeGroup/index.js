import LayersTreeGroup from './LayersTreeGroup';

export * from './LayersTreeGroup';
export default LayersTreeGroup;
