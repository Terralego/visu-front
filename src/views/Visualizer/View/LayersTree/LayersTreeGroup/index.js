import { connectAuthProvider } from '@terralego/core/modules/Auth';

import LayersTreeGroup from './LayersTreeGroup';

export default connectAuthProvider('authenticated')(LayersTreeGroup);
