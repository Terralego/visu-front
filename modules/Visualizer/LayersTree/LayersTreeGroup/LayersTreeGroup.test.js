import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';

import LayersTreeGroup from './LayersTreeGroup';

jest.mock('@blueprintjs/core', () => ({
  H5 ({ children }) {
    return children;
  },
  Button () {
    return <p>Button</p>;
  },
  Collapse ({ children }) {
    return children;
  },
}));

jest.mock('../LayersTreeItem', () => function LayersTreeItem () {
  return <p>LayersTreeItem</p>;
});

it('should render correctly', () => {
  const tree = renderer.create((
    <>
      <LayersTreeGroup
        title="Group 1"
        layer={{
          layers: [{
            label: 'Layer 1',
          }, {
            label: 'Layer 2',
          }],
        }}
      />
      <LayersTreeGroup
        title="Group 1"
        layer={{
          layers: [{
            label: 'Layer 1',
          }, {
            label: 'Layer 2',
          }],
        }}
        isHidden
      />
    </>
  )).toJSON();
  expect(tree).toMatchSnapshot();
});

it('should change layers group state', () => {
  const wrapper = shallow((
    <LayersTreeGroup
      title="Group 1"
      layer={{
        layers: [{
          label: 'Layer 1',
        }, {
          label: 'Layer 2',
        }],
      }}
      isOpen
    />
  ));
  expect(wrapper.state().isOpen).toEqual(true);
  wrapper.setState({
    isOpen: false,
  });
  expect(wrapper.state().isOpen).toEqual(false);
});

it('should open group panel', () => {
  const instance = new LayersTreeGroup({});
  instance.setState = jest.fn();

  instance.state = {
    isOpen: true,
  };
  instance.handleClick();
  expect(instance.setState).toHaveBeenCalledWith({
    isOpen: false,
  });
  instance.setState.mockClear();

  instance.state = {
    isOpen: false,
  };
  instance.handleClick();
  expect(instance.setState).toHaveBeenCalledWith({
    isOpen: true,
  });
});
