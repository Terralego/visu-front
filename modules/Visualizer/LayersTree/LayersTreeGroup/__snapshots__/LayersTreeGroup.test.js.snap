// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
Array [
  <div
    className="layerstree-group layerstree-group--0 layerstree-group--active"
  >
    <p>
      Button
    </p>
    <p>
      LayersTreeItem
    </p>
    <p>
      LayersTreeItem
    </p>
  </div>,
  <div
    className="layerstree-group layerstree-group--0"
  >
    <p>
      Button
    </p>
    <p>
      LayersTreeItem
    </p>
    <p>
      LayersTreeItem
    </p>
  </div>,
]
`;
